components/backup-stream/src/router.rs

```diff
@@ -1286,6 +1296,14 @@ mod tests {
             .expect("failed to register task")
     }
 
+    fn check_on_events_result(item: &Vec<(String, Result<()>)>) {
+        for (task, r) in item {
+            if let Err(err) = r {
+                panic!("task {} failed: {}", task, err);
```

Comment:
Panic when on_events failed ?
Can we have a better way ?

Reply:
This function would only be called in tests, the real handler for the result of `on_event` is in the `utils::handle_on_event_result`, which sends a `FatalError` to the endpoint.